pkg/server/api/agent/v1/service.go

```diff
@@ -250,21 +230,18 @@ func (s *Service) RenewAgent(ctx context.Context, req *agent.RenewAgentRequest)
 	log := rpccontext.Logger(ctx)
 
 	if err := rpccontext.RateLimit(ctx, 1); err != nil {
-		log.WithError(err).Error("Rejecting request due to renew agent rate limiting")
-		return nil, err
+		return nil, api.MakeErr(log, codes.Unavailable, "rejecting request due to renew agent rate limiting", err)
```

Comment:
Maybe `codes.ResourceExhausted` is a better fit for rate limiting errors?

Reply:
do you verify what error is returned by rate limit? i can recall it returned a grpc status with the correct code

Reply:
I was wondering the same... I am not sure which one to choose. I found precedence for both of them:

+ The [HTTP to gRPC mapping table](https://github.com/grpc/grpc/blob/master/doc/http-grpc-status-mapping.md) suggests mapping code `429` `Too Many Requests` to `UNAVAILABLE`.

+ The [Google cloud API design guide](https://developers.google.com/maps-booking/reference/grpc-api/status_codes) suggests using `RESOURCE_EXAUSTED`

Reply:
I did not see your previous comment @MarcosDY. Now that you mention it, it seems to me that the right thing to do is to extract and use the error code that comes from the rate limiter. Which could be: Internal,  ResourceExausted or InvalidArgument.

Reply:
or we can just use error from limiter as it was